galley/pkg/runtime/processor.go

```diff
@@ -156,13 +138,9 @@ loop:
 
 		// Incoming events are received through p.events
 		case e := <-p.events:
-			scope.Debugf("Processor.process: event: %v", e)
-			if p.processEvent(e) {
-				scope.Debugf("Processor.process: event: %v, signaling onChange", e)
-				p.strategy.onChange()
-			}
+			_ = p.processEvent(e)
 
-		case <-p.strategy.publish:
+		case <-p.state.strategy.publish:
```

Comment:
@ozevren @ayj would it be better to give each state own process loop? If we did that, there would be a second channel for events that goes from processor -> state. 

If we had this in place, state could have start/stop functions ... it would effectively be a processor ... and then processor would be a composite processor.